src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -23,6 +23,44 @@
 namespace openassetio {
 inline namespace OPENASSETIO_CORE_ABI_VERSION {
 namespace {
+
+/**
+ * Validate the supplied ManagerInterface supports all required
+ * capabilities, or throw a ConfigurationException.
+ *
+ * These capabilities are required at runtime, but not enforced at
+ * compile time, as multi-language implementations may only provide the
+ * implementation in one of the component languages.
+ */
+void verifyRequiredCapabilities(const managerApi::ManagerInterfacePtr &interface) {
+  using managerApi::ManagerInterface;
+  using CapabilityVector = std::vector<ManagerInterface::Capability>;
+
+  static const CapabilityVector kRequiredCapabilities = {
+      ManagerInterface::Capability::kEntityReferenceIdentification,
+      ManagerInterface::Capability::kManagementPolicyQueries};
+
+  CapabilityVector missingCapabilities;
+  for (const ManagerInterface::Capability capability : kRequiredCapabilities) {
+    if (!interface->hasCapability(capability)) {
+      missingCapabilities.push_back(capability);
+    }
+  }
+
+  if (missingCapabilities.empty()) {
+    return;
+  }
+
+  std::string msg =
+      fmt::format("Manager implementation for '{}' does not support the required capabilities:",
+                  interface->identifier());
+  for (const ManagerInterface::Capability capability : missingCapabilities) {
+    msg += fmt::format(" {}", ManagerInterface::kCapabilityNames[static_cast<size_t>(capability)]);
+  }
```

Comment:
[Consider][minor]
This implementation is perfectly fine. You might consider extracting the string conversion from the message construction though, it seems cleaner to me, makes it clear that the intent is to have a comma (or space) seperated list, but this is very much a personal style thing.
:
```suggestion
  // Convert missing capabilities to their string representations
  std::vector<std::string> missingCapabilityStrings;
  std::transform(missingCapabilities.cbegin(), missingCapabilities.cend(), 
  std::back_inserter(missingCapabilityStrings), [](const ManagerInterface::Capability cap) {
      return ManagerInterface::kCapabilityNames[static_cast<size_t>(capability)]; 
 });
  
  // Construct error message
  std::string msg =
      fmt::format("Manager implementation for '{}' does not support the required capabilities: {}",
                  interface->identifier(), fmt::join(missingCapabilityStrings, ", ")););
  }
```

Reply:
Ooooooh, I like the join, lets me do what I wanted to in the first place with a comma. Updated based on this [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/1156/commits/90a0e40cdf4fd45e4a7143a26a2ae404b4fc7e67)

Reply:
Oh yeah just doing it with strings in the first pass, makes hella sense.

Reply:
Teamwork for the win!!